### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by wale-prog**
In this pull request, I have implemented the following:
- Created class `Classroom` with `@label` instance variable initialized with the constructor
- Created `has-many` / `belongs-to` relationship between `Classroom` and `Student`.
- Created a class `Book` with `@title` and `@author` instance variables initialized with the constructor
- Created class `Rental` with `@date` instance variable initialized with the constructor.
- Created `has-many` / `belongs-to` and (also `has-many-through`) relationship between `Person` and `Book`.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
```
Comment (t-yanick) [inline book.rb]: - [ ] The book `add_rental` method should also contain the date parameter, some like so: 
```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```
Please kindly fix that.

File: person.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (t-yanick) [inline person.rb:16]: - [ ] To implement the `has-many` side of the relationship(a person has many rentals), a method for adding rentals should be present, something like so: 

```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```
Please kindly fix that.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @wale-prog, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 1 above)
Comment (Emmyn5600) [reply book.rb]: Well solved

File: person.rb (hunk 2 above)
Comment (Emmyn5600) [reply person.rb:16]: Well solved

**Review by Emmyn5600 — APPROVED** [review]
Hi @wale-prog,

Your project is complete!  There is nothing else to say other than... it's time to merge it :squirrel:

Congratulations! :tada:

Cheers and Happy coding!:clap::clap::clap:

##### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification.
------
###### _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**